File: frontend/assistant-on-nextjs/src/lib/api/chemistry/CompoundGraphApi.ts
import { ChemistryGraphResponse } from "./graphModels"

export default interface CompoundGraphApi {
  doPubChemGraphGetAll(page: number, pageSize: number): Promise<ChemistryGraphResponse>

  doPubChemGraphGetCompoundDataByName(name: string, page: number, pageSize: number): Promise<ChemistryGraphResponse>
}

File: java/assistant-on-spring-boot-servlet/app/src/main/java/org/example/assistantonsbservlet/api/chemistry/graph/CompoundGraphApi.java
package org.example.assistantonsbservlet.api.chemistry.graph;

import org.example.assistantonsbservlet.api.ChemistryGraphResponse;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

@RequestMapping("/api/v1/chemistry/graph/compound")
public interface CompoundGraphApi {
    @GetMapping(params = "name")
    ResponseEntity<ChemistryGraphResponse> getCompoundDataByName(
        @RequestParam(defaultValue = "1") int page,
        @RequestParam(defaultValue = "10") int pageSize,
        @RequestParam String name
    );

    @GetMapping
    ResponseEntity<ChemistryGraphResponse> getAll(
        @RequestParam(defaultValue = "1") int page,
        @RequestParam(defaultValue = "10") int pageSize
    );
}
